ml-dhs-aws/src/main/java/com/marklogic/client/dhs/CountInputStream.java

```diff
@@ -0,0 +1,32 @@
+package com.marklogic.client.dhs;
+
+import java.io.FileNotFoundException;
+import java.io.FilterInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+public class CountInputStream extends FilterInputStream {
+	public CountInputStream(InputStream inputStream) throws FileNotFoundException {
+		super(inputStream);
+		this.byteCount = 0;
+	}
+
+	private int byteCount;
+
+	public int getByteCount() {
+		try {
+			read();
+		} catch (IOException ex) {
+			ex.printStackTrace();
+		}
+		return byteCount;
+	}
+
+	@Override
+	public int read() throws IOException {
+		int readLength = super.read();
+		byteCount += readLength;
+		return readLength;
+	}
+
```

Comment:
Because the consumer may use any read() method to access the wrapped input stream, CountInputStream must override all of the read() method overloads, capturing the count returned from the call to the equivalent super.read() overload, incrementing byteCount, and then returning the count.